### ISSUE TITLE
WS Subscription Events Silently Dropped on Parse Error

### ISSUE THREAD

**Issue by ryanrossiter** [informalsystems/tendermint-rs#1399]
### What went wrong?

When using the WebSocketClient to subscribe to events, if the event cannot be parsed into a JSON RPC event due to a deserialization error, the event is [silently ignored](https://github.com/informalsystems/tendermint-rs/blob/main/rpc/src/client/transport/websocket.rs#L923). This is then difficult to debug why events aren't being received.

### Steps to reproduce

Subscribe to a TM 37 chain using the TM 34 compatibilty mode, which will cause event parsing to fail since the events won't be decodable from base64.

### Definition of "done"

Event parsing errors should be relayed to the subscriber to handle.

**Comment by romac** [comment]
Agreed, we should indeed forward such errors to subscribers. Would you be willing to open a PR to address this?
